lib/validateToken.js

```diff
@@ -1,75 +1,33 @@
-var crypto = require("crypto");
-var request = require('request');
-var defer = require('promised-io/promise').defer;
-var when = require("promised-io/promise").when;
-var utils = require('./utils');
-
-var userIdRegex = /un=(\w+\@\w+(\.\w+))/;
-var ss_cache = {};
-
-getSigner = function(signer){
-    var def = new defer();
-    if (ss_cache[signer]){
-        def.resolve(ss_cache[signer]);
-    }
-    request.get({url:signer,json:true}, function(err,response,body){
-        if (err) { return def.reject(err); }
-        if (!body) { return def.reject("Empty Signature"); }
-        def.resolve(body.pubkey);
-    });
-    return def.promise;
-};
-
-var validateToken = function(token){
-    var parts = token.split("|");
-    var parsedToken = {};
-    var baseToken = [];
-    parts.forEach(function(part){
-        var tuple = part.split("=");
-        if (tuple[0] !== "sig"){
-            baseToken.push(part);
+'use strict'
+
+var Request = require('request-promise')
+/**
+ * Given an auth token, fetch the user session information from
+ * the Auth service.
+ */
+var getUserSession = function(authHost, token) {
+    return Request({
+        url: authHost + '/api/V2/me',
+        method: 'GET',
+        headers: {
+            'Authorization': token
         }
-        parsedToken[tuple[0]]=tuple[1];
-    });
-
-
-    var ssString = parsedToken.SigningSubject,
-    signingSubject = ssString.slice(0, ssString.lastIndexOf('/'));
-
-    if (signingSubject !== "https://nexus.api.globusonline.org/goauth/keys") {
-        return false;
-    }
-
-
-    return when(getSigner(parsedToken.SigningSubject), function(signer){
-        var verifier = crypto.createVerify("RSA-SHA1");
-        verifier.update(baseToken.join("|"));
-        var success = verifier.verify(signer.toString("ascii"),parsedToken.sig,"hex");
-        return success;
-    }, function(err) {
-        utils.log('ERROR', "Error retrieving SigningSubject: ", parsedToken.SigningSubject);
-        return false;
-    });
-
-};
-
-module.exports = function(token) {
-    return when(validateToken(token), function(valid){
-        if (!valid) {
-            utils.log('ERROR', "Invalid Token");
-            return false;
-        }
-
-        var user = {
-            id: token.split('|')[0].replace('un=', ''),
-            token: token
-        };
-
-        // console.log("User from token: ", user.id);
-        if (user && user.id) {
-            return user;
-        }
-        return false;
-    });
-
-};
+    }).then(session => {
+        session = JSON.parse(session)
+        session.token = token
+        var globusUser = null
+        // find globus username(s) if any, and put *JUST THE USERNAME* in the globusUser key
+        // otherwise, make it null
+        session.idents.forEach(ident => {
+            if (ident.provider === 'Globus') {
+                globusUser = ident.username
```

Comment:
Right now it just picks one and puts it there. I'm not sure what the right behavior is, or should be, honestly. It could drop all identities in that file, but that won't help much.